cluster-autoscaler/cloudprovider/mcm/mcm_cloud_provider.go

```diff
@@ -371,10 +376,66 @@ func (machinedeployment *MachineDeployment) DeleteNodes(nodes []*apiv1.Node) err
 		}
 		ref, err := ReferenceFromProviderID(machinedeployment.mcmManager, node.Spec.ProviderID)
 		if err != nil {
-			return fmt.Errorf("Couldn't find the machine-name from provider-id %s", node.Spec.ProviderID)
+			return fmt.Errorf("couldn't find the machine-name from provider-id %s", node.Spec.ProviderID)
 		}
 		machines = append(machines, ref)
 	}
+
+	retryDeadline := time.Now().Add(maxRetryTimeout)
+	for {
+		updatedMachines := true
+		machinesClient := machinedeployment.mcmManager.machineClient.Machines(machinedeployment.mcmManager.namespace)
+		allMachines, err := machinesClient.List(context.TODO(), v1.ListOptions{LabelSelector: labels.FormatLabels(map[string]string{machineDeploymentNameLabel: machinedeployment.Ref.Name})})
+		if err != nil {
+			return fmt.Errorf("couldn't list all machines for machine deployment %s, error: %v", machinedeployment.Ref.Name, err)
+		}
+		for _, machine := range allMachines.Items {
+			if machine.Labels != nil {
+				nodeName := machine.Labels[v1alpha1.NodeLabelKey]
+				node, _ := machinedeployment.mcmManager.nodeLister.Get(nodeName)
+				if node != nil {
+					flag := 0
+					for _, taint := range node.Spec.Taints {
+						if taint.Key == deletetaint.ToBeDeletedTaint {
+							flag = 1
+							break
+						}
+					}
+					if flag == 1 {
+						// Don't update priority annotation if the taint is present
+						continue
+					}
+				}
+			}
+			annotations := make(map[string]string, len(machine.Annotations))
+			flag := 0
+			for key, val := range machine.Annotations {
+				if key == machinePriorityAnnotation && val == "1" {
+					flag = 1
+					annotations[machinePriorityAnnotation] = "3"
+				} else {
+					annotations[key] = val
+				}
+			}
+			if flag == 1 {
+				clone := machine.DeepCopy()
+				clone.Annotations = annotations
+				_, err := machinesClient.Update(context.TODO(), clone, v1.UpdateOptions{})
+				if err != nil && time.Now().Before(retryDeadline) {
+					klog.Warningf("could not update priority annotation on existing machine %s; will retry this operation, error: %v", clone.Name, err)
```

Comment:
kindly also mention the time in which you'll retry , easier to track in logs then
```suggestion
klog.Warningf("could not reset priority annotation for machine %s; will retry this operation, error: %v", clone.Name, err)
```

Reply:
kindly commit all these suggestions